### PR TITLE
709 update navbar picture

### LINKED ISSUE #709: Update navbar when user changes profile picture
Removing the picture updates the navbar but changing it doesn't.
Do the same trick.

### PR DESCRIPTION
**Pull request by ArsalanSavand**
# Bug fixed
Fixes #709

### CODE REVIEW

File: app/header/header_controller.js [hunk 1]
```diff
@@ -149,8 +149,10 @@ function HeaderController($scope, $rootScope, $state, $stateParams,
   /**
    * @desc User update
    */
-  $scope.$on("gonevisDash.UserController:update", constructor);
-
+  $scope.$on("gonevisDash.UserController:update", function (event, data) {
+    $scope.user = AuthService.setAuthenticatedUser(data, true);
+  });
```
Comment (AmirSavand) [inline app/header/header_controller.js]: The header controller should not handle updating user, the controller that is responsible for it should, in this case, user controller.

So update the user with `setAuthenticatedUser` then broadcast `gonevisDash.UserController:update` and revert these lines, constructor should be called.

File: app/header/header_controller.js [hunk 2]
```diff
@@ -165,6 +167,7 @@ HeaderController.$inject = [
   "DolphinService",
   "Codekit",
   "Entry",
+  "Account",
```
Comment (AmirSavand) [inline app/header/header_controller.js]: Don't forget to remove this.

File: app/header/header_controller.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 "use strict";
 
 function HeaderController($scope, $rootScope, $state, $stateParams,
-  AuthService, DolphinService, Codekit, Entry, API, ModalsService, toaster) {
+  AuthService, DolphinService, Codekit, Entry, Account, API, ModalsService, toaster) {
```
Comment (AmirSavand) [inline app/header/header_controller.js]: Remove.

File: app/account/user/user_controller.js [hunk 4]
```diff
@@ -41,7 +41,7 @@ function UserController($scope, $rootScope, $stateParams,
           $scope.user[key] = data[key];
         }
         $scope.user = AuthService.setAuthenticatedUser($scope.user.get, true);
-        $rootScope.$broadcast("gonevisDash.UserController:update");
+        $rootScope.$broadcast("gonevisDash.UserController:update", data);
```
Comment (AmirSavand) [inline app/account/user/user_controller.js]: No need to pass data.

File: app/account/user/user_controller.js [hunk 5]
```diff
@@ -112,6 +112,7 @@ function UserController($scope, $rootScope, $stateParams,
     }).then(function (data) {
       toaster.info("Done", "Profile picture updated");
       $scope.user = new Account(data.data);
+      $rootScope.$broadcast("gonevisDash.UserController:update", data.data);
```
Comment (AmirSavand) [inline app/account/user/user_controller.js]: Update user data, that's enough.

File: app/header/header_controller.js (hunk 3 above)
Comment (ArsalanSavand) [inline app/header/header_controller.js]: Done

File: app/header/header_controller.js (hunk 1 above)
Comment (ArsalanSavand) [inline app/header/header_controller.js]: Done

File: app/header/header_controller.js (hunk 2 above)
Comment (ArsalanSavand) [inline app/header/header_controller.js]: Done

File: app/account/user/user_controller.js (hunk 4 above)
Comment (ArsalanSavand) [inline app/account/user/user_controller.js]: Done

File: app/account/user/user_controller.js (hunk 5 above)
Comment (ArsalanSavand) [inline app/account/user/user_controller.js]: Done

**Pull request merged**